libs/application/template-api-modules/src/lib/modules/templates/party-application/party-application.service.ts

```diff
@@ -0,0 +1,103 @@
+import { TemplateApiModuleActionProps } from '../../../types'
+import { SharedTemplateApiService } from '../../shared'
+import {
+  generateAssignSupremeCourtApplicationEmail,
+  generateApplicationRejectedEmail,
+  generateApplicationApprovedEmail,
+} from './emailGenerators'
+import { Constituencies } from '@island.is/application/templates/party-application'
+import { EndorsementListTagsEnum } from './gen/fetch'
+
+type ErrorResponse = {
+  errors: {
+    message: string
+  }
+}
+
+type CreateEndorsementListResponse =
+  | {
+      data: {
+        endorsementSystemCreateEndorsementList: {
+          id: string
+        }
+      }
+    }
+  | ErrorResponse
+
+const constituencyMapper: Record<Constituencies, EndorsementListTagsEnum> = {
+  [Constituencies.NORTH_EAST]:
+    EndorsementListTagsEnum.partyApplicationNordausturkjordaemi2021,
+  [Constituencies.NORTH_WEST]:
+    EndorsementListTagsEnum.partyApplicationNordvesturkjordaemi2021,
+  [Constituencies.RVK_NORTH]:
+    EndorsementListTagsEnum.partyApplicationReykjavikurkjordaemiNordur2021,
+  [Constituencies.RVK_SOUTH]:
+    EndorsementListTagsEnum.partyApplicationReykjavikurkjordaemiSudur2021,
+  [Constituencies.SOUTH]:
+    EndorsementListTagsEnum.partyApplicationSudurkjordaemi2021,
+  [Constituencies.SOUTH_WEST]:
+    EndorsementListTagsEnum.partyApplicationSudvesturkjordaemi2021,
+}
+
+@Injectable()
+export class PartyApplicationService {
+  constructor(
+    private readonly sharedTemplateAPIService: SharedTemplateApiService,
+  ) {}
+
+  async assignSupremeCourt({ application }: TemplateApiModuleActionProps) {
+    await this.sharedTemplateAPIService.assignApplicationThroughEmail(
+      generateAssignSupremeCourtApplicationEmail,
+      application,
+    )
+  }
+
+  async applicationRejected({ application }: TemplateApiModuleActionProps) {
+    await this.sharedTemplateAPIService.sendEmail(
+      generateApplicationRejectedEmail,
+      application,
+    )
+  }
+
+  async applicationApproved({ application }: TemplateApiModuleActionProps) {
+    await this.sharedTemplateAPIService.sendEmail(
+      generateApplicationApprovedEmail,
+      application,
+    )
+  }
+
+  async createEndorsementList({
+    application,
+    authorization,
+  }: TemplateApiModuleActionProps) {
+    const constituencyTag =
+      constituencyMapper[application.answers.constituency as Constituencies]
+    const CREATE_ENDORSEMENT_LIST_QUERY = `
+      mutation {
+        endorsementSystemCreateEndorsementList(input: {
+          title: "${application.answers.partyName}",
+          description: "${application.answers.partyLetter}",
+          endorsementMeta: [fullName, address, signedTags],
+          tags: [${constituencyTag}],
+          validationRules: [],
+        }) {
+          id
+        }
+      }
+    `
+
+    const endorsementList: CreateEndorsementListResponse = await this.sharedTemplateAPIService
+      .makeGraphqlQuery(authorization, CREATE_ENDORSEMENT_LIST_QUERY)
```

Comment:
Mind adding a new arg to the makeGraphlQuery so we can pass the input there rather than string interpolation. e.g.

- https://github.com/island-is/island.is/blob/main/libs/application/core/src/types/BasicDataProvider.ts#L66
- https://github.com/island-is/island.is/blob/main/libs/application/templates/health-insurance/src/dataProviders/ApplicationsProvider.ts#L13-L24

Reply:
Yes, will update